src/libraries/System.IO.FileSystem/tests/FileStream/ReadAsync.cs

```diff
@@ -101,7 +101,7 @@ public async Task ReadAsyncCanceledFile(int bufferSize, bool isAsync)
             }
         }
 
-        [ConditionalTheory(typeof(PlatformDetection), nameof(PlatformDetection.IsThreadingSupported))]
+        [ConditionalTheory(typeof(PlatformDetection), nameof(PlatformDetection.IsThreadingSupported), nameof(PlatformDetection.IsNet5CompatFileStreamDisabled))]
```

Comment:
Why is this needed?

Reply:
I'd also expect test to pass regardless of any compat switches. The culprit seems to be that we reused the same buffer for parallel reads

Reply:
This test from time to time fails when .NET 5 Compat is enabled:

From sample [log](https://helixre8s23ayyeko0k025g8.blob.core.windows.net/dotnet-runtime-refs-pull-57271-merge-faf84f53978341e4ae/System.IO.FileSystem.Net5Compat.Tests/1/console.b7329ad5.log?sv=2019-07-07&se=2021-09-01T09%3A48%3A48Z&sr=c&sp=rl&sig=qwhFEhF2JXTSNwfpAsgJSKuDfiayTcAtTTWovXpLvHg%3D):

```log
    System.IO.Tests.FileStream_BeginEndRead_AsyncReads.IncompleteReadCantSetPositionBeyondEndOfFile(fileShare: None, options: Asynchronous) [FAIL]
      Showing first 10 differences
        Position 0: Expected: 33, Actual: 0
        Position 1: Expected: 174, Actual: 0
        Position 2: Expected: 151, Actual: 0
        Position 3: Expected: 146, Actual: 0
        Position 4: Expected: 163, Actual: 0
        Position 5: Expected: 195, Actual: 0
        Position 6: Expected: 16, Actual: 0
        Position 7: Expected: 106, Actual: 0
        Position 8: Expected: 142, Actual: 0
        Position 9: Expected: 125, Actual: 0
      Total number of differences: 9968 out of 10000
      Stack Trace:
        /_/src/libraries/Common/tests/TestUtilities/System/AssertExtensions.cs(473,0): at System.AssertExtensions.SequenceEqual[T](ReadOnlySpan`1 expected, ReadOnlySpan`1 actual)
        /_/src/libraries/Common/tests/TestUtilities/System/AssertExtensions.cs(491,0): at System.AssertExtensions.SequenceEqual[T](Span`1 expected, Span`1 actual)
        /_/src/libraries/System.IO.FileSystem/tests/FileStream/ReadAsync.cs(128,0): at System.IO.Tests.FileStream_AsyncReads.IncompleteReadCantSetPositionBeyondEndOfFile(FileShare fileShare, FileOptions options)
        --- End of stack trace from previous location ---
  Finished:    System.IO.FileSystem.Net5Compat.Tests
```

Since it's a new test that I've added recently and it's not a .NET 5 regression, I did not loook into that.